addons/l10n_jo_edi/models/account_move.py

```diff
@@ -0,0 +1,159 @@
+            raise ValidationError("Client ID is missing.\n To set it: Configuration > Settings > Electronic Invoicing (Jordan)")
+        if not self.company_id.l10n_jo_edi_secret_key:
+            raise ValidationError("Secret key is missing.\n To set it: Configuration > Settings > Electronic Invoicing (Jordan)")
+        return {
+            'Client-Id': self.company_id.l10n_jo_edi_client_id,
+            'Secret-Key': self.company_id.l10n_jo_edi_secret_key,
+            'Content-Type': 'application/json',
+        }
+
+    def submit_to_jofotara(self):
+        self.ensure_one()
+        headers = self._build_jofotara_headers()
+        params = '{"invoice": "%s"}' % self.l10n_jo_edi_xml_attachment_id.datas.decode('ascii')
+
+        try:
+            response = requests.post(JOFOTARA_URL, data=str(params), headers=headers, timeout=50, verify=False)
+        except (requests.exceptions.Timeout, requests.exceptions.RequestException, requests.exceptions.HTTPError) as e:
+            raise UserError("Invalid request: %s" % e)
+
+        response_text = response.content.decode('utf-8')
+        if not response.ok:
+            raise UserError("Request failed: %s" % response_text)
+        dict_response = json.loads(response_text)
+        self.l10n_jo_edi_qr = str(dict_response.get('EINV_QR', ''))
+
+    def _l10n_jo_edi_get_xml_attachment_name(self):
+        return f"{self.name.replace('/', '_')}_edi.xml"
+
+    def _l10n_jo_edi_send(self):
+        for invoice in self:
+            invoice.l10n_jo_edi_xml_attachment_id.res_field = False
+            invoice_xml = self.env["ir.attachment"].create(
+                {
+                    "res_model": "account.move",
+                    "res_id": invoice.id,
+                    "res_field": "l10n_jo_edi_xml_attachment_file",
+                    "name": invoice._l10n_jo_edi_get_xml_attachment_name(),
+                    "datas": base64.b64encode(invoice.env['account.edi.xml.ubl_21.jo']._export_invoice(invoice)),
+                }
+            )
+            invoice.invalidate_recordset(
+                fnames=[
+                    "l10n_jo_edi_xml_attachment_id",
+                    "l10n_jo_edi_xml_attachment_file",
+                ]
+            )
+
+            try:
+                invoice.submit_to_jofotara()
+                invoice.l10n_jo_edi_error = False
+                invoice.with_context(no_new_invoice=True).message_post(
+                    body=_("E-invoice (JoFotara) submitted successfully."),
+                    attachment_ids=invoice_xml.ids,
+                )
+            except (ValidationError, UserError) as error:
+                invoice.l10n_jo_edi_error = error
+                invoice.with_context(no_new_invoice=True).message_post(
+                    body=_("E-invoice (JoFotara) submission failed. %s", error),
+                    attachment_ids=invoice_xml.ids,
+                )
+                return error
```

Comment:
The idea would be to pass what is in the message_post to error.  (passing the error object itself might be a bit harsh)